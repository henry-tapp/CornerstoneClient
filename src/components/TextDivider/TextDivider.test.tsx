import React from "react";

import { render, screen } from "@testing-library/react";

import { TextDivider } from "./";

it("renders without crashing", () => {
  const { asFragment } = render(<TextDivider text="OR" />);

  expect(screen.getByTestId("text-divider-left")).toBeInTheDocument();
  expect(screen.getByTestId("text-divider-right")).toBeInTheDocument();
  expect(screen.getByText(/OR/i)).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});

it("only renders a single hr if no text provided", () => {
  const { asFragment } = render(<TextDivider />);

  expect(screen.getByTestId("text-divider-left")).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
