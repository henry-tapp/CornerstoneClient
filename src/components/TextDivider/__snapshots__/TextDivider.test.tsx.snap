// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`only renders a single hr if no text provided 1`] = `
<DocumentFragment>
  <div
    class="text-divider "
    data-testid="text-divider-wrapper"
  >
    <hr
      data-testid="text-divider-left"
    />
  </div>
</DocumentFragment>
`;

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="text-divider "
    data-testid="text-divider-wrapper"
  >
    <hr
      data-testid="text-divider-left"
    />
    <span>
      OR
    </span>
    <hr
      data-testid="text-divider-right"
    />
  </div>
</DocumentFragment>
`;
